mwtext/utilities/preprocess_wikidata.py

```diff
@@ -34,25 +62,40 @@ def isRelevantEntity(entity):
     return (len(sitelinks) > 0)
 
 
-def preprocess_wikidata(WIKIDATA_DIR):
-    with bz2.open(os.path.join(WIKIDATA_DIR, 'latest-all.json.bz2'), 'rt') as fin:
-        next(fin)
-        for idx, line in enumerate(fin, start=1):
-            try:
-                item_json = json.loads(line)
-            except Exception:
-                try:
-                    item_json = json.loads(line[:-2])
-                except Exception:
-                    print("Error:", idx)
-                    continue
-
-            qid = item_json.get('id', None)
-            if not qid:
-                continue
-
-            entity = mwbase.Entity.from_json(item_json)
-            if isRelevantEntity(entity):
-                prop_val = getPropertyValue(entity)
-                yield qid + ": " + prop_val
+def preprocess_wikidata(dump):
+    wikidata_preprocessor = WikidataPreprocessor()
+    for item_json in dump:
```

Comment:
we already have wikidata item in the required format here. wikidata_preprocessor.process() will just yield the same wikidata item that has been passed to it?

Reply:
It should yield a sequence of tokens that we want to build embeddings over.  This would be the Qids and Pids we were trying to extract.

Reply:
got it! I have made the changes